File: SNACKS/Models/IngresoInsumo.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace SNACKS.Models
{
    public class IngresoInsumo
    {
        public int IdIngresoInsumo { get; set; }
        public Usuario Usuario { get; set; }
        public Almacen Almacen { get; set; }
        public Caja Caja { get; set; }
        public DateTime FechaCreacion { get; set; }
        public string Comentario { get; set; }
        public decimal Costo { get; set; }
        public List<ItemIngresoInsumo> Items { get; set; }

        public IngresoInsumo()
        {
            Items = new List<ItemIngresoInsumo>();
        }
    }
}

File: SNACKS/ClientApp/src/app/ingresos-insumo/ingreso-insumo.ts
import { IProducto } from "../productos/producto";
import { IUnidad } from "../unidades/unidad";
import { IUsuario } from "../usuarios/usuario";
import { IAlmacen } from "../almacenes/almacen";
import { ICaja } from "../cajas/caja";

export interface IIngresoInsumo {
  idIngresoInsumo: number;
  fechaCreacion: Date;
  usuario: IUsuario;
  comentario: string;
  costo: number;
  almacen: IAlmacen,
  caja: ICaja,
  items: IItemIngresoInsumo[];
}

export interface IItemIngresoInsumo {
  idItemIngresoInsumo?: number;
  ingresoInsumo?: IIngresoInsumo;
  producto?: IProducto;
  unidad?: IUnidad;
  factor?: number;
  cantidad?: number;
  costo?: number;
}
